notes_app/client/notes.js

```diff
@@ -45,15 +45,10 @@
       'data': {
         'loginRequired': true
       },
-      'onEnter': function($state, userService, $q) {
-          var user = $q(function(resolve,reject){
-            resolve(userService.user);
-          });
-          user.then(function(response){
-            if (!response.data.is_superuser ) {
-                $state.go('main.note.list');
-            }
-          })
+      'onEnter': function($state, user) {
+        if (!user.data.is_superuser || !user.logged_in) {
```

Comment:
i have checked it for `is_superuser` at first but it turned out that it's not enough.